src/test/java/org/ibp/api/java/impl/middleware/inventory/manager/LotServiceImplTest.java

```diff
@@ -206,4 +240,97 @@ public void shouldSplitLot() {
         assertThat(depositLotLotIds, contains(newSplitExtendedLotDto.getLotId()));
     }
 
+    @Test
+    public void testCreateLots_ThrowsException_When_SearchOriginIsNull() {
+        final String keepLotUUID = "keepLotUUID";
+        final LotGeneratorBatchRequestDto lotGeneratorBatchRequestDto = new LotGeneratorBatchRequestDto();
+        final LotGeneratorInputDto lotGeneratorInputDto = new LotGeneratorInputDto();
+        final SearchCompositeDto searchCompositeDto = new SearchCompositeDto<>();
+        final SearchOriginCompositeDto searchOriginCompositeDto = new SearchOriginCompositeDto();
+        searchOriginCompositeDto.setSearchOrigin(null);
+        searchOriginCompositeDto.setSearchRequestId(new Random().nextInt());
+
+        searchCompositeDto.setSearchRequest(searchOriginCompositeDto);
+        lotGeneratorInputDto.setStockPrefix(UUID.randomUUID().toString());
+        lotGeneratorInputDto.setLocationId(new Random().nextInt());
+
+        lotGeneratorBatchRequestDto.setLotGeneratorInput(lotGeneratorInputDto);
+        lotGeneratorBatchRequestDto.setSearchComposite(searchCompositeDto);
+        try {
+        this.lotService.createLots(keepLotUUID, lotGeneratorBatchRequestDto);
+        } catch (final ApiRequestValidationException e) {
+            assertThat(Arrays.asList(e.getErrors().get(0).getCodes()), hasItem("search.origin.no.defined"));
+        }
+    }
+
+    @Test
+    public void testCreateLots_ThrowsException_WhenForGermplasmsearchTheSearchRequestIdNoReturnGids() {
+        final String keepLotUUID = "keepLotUUID";
+        final LotGeneratorBatchRequestDto lotGeneratorBatchRequestDto = new LotGeneratorBatchRequestDto();
```

Comment:
I think you could have a private function to build the lotGeneratorBatchRequestDto passing some params and the code will be drastically reduced.

Reply:
done.